api/v1alpha1/envoyproxy_types.go

```diff
@@ -86,8 +86,45 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
+
+	// FilterOrder defines the order of filters in the Envoy proxy's HTTP filter chain.
+	FilterOrder *FilterOrder `json:"filterOrder,omitempty"`
+}
+
+// FilterOrder defines the order of filters in the Envoy proxy's HTTP filter chain.
+// If unspecified, the default order of filters is applied.
+// Default order of filters:
+// - envoy.filters.http.cors
+// - envoy.filters.http.ext_authz
+// - envoy.filters.http.basic_authn
+// - envoy.filters.http.oauth2
+// - envoy.filters.http.jwt_authn
+// - envoy.filters.http.fault
+// - envoy.filters.http.local_ratelimit
+// - envoy.filters.http.rate_limit
+type FilterOrder struct {
+	// EnvoyFilters defines the order of filters in the filter chain.
+	// The order of filters in the list is the order in which they will be applied.
+	// +kubebuilder:validation:MinItems=8
```

Comment:
How would this behave when users upgrade to a new version that includes an additional filter? I assume that we will increase min/max accordingly, which will make the existing end-user configuration invalid, causing things like gitops that applies user-defined resources to fail. Users that choose to change the order of some filters now need to know about all filters. 

Maybe we can avoid min/max here? Some ideas:
- Partial list will only impact ordering of the mentioned filters between themselves
- Instead of filter names, use a tuple of <filterName, filterPriority>, allowing users to control order specifically for filters. Native order will use 100-unit increments (0, 100, 200, ... ) to allow users plenty of room to re-order filters in-between the natively ordered filters

Reply:
> Instead of filter names, use a tuple of <filterName, filterPriority>, allowing users to control order specifically for desired filters. Native order will use 100-unit increments (0, 100, 200, ... ) to allow users plenty of room to re-order filters in-between the natively ordered filters

Vote for this option because users normally only need to adjust one filter or two. They don't have to care about the orders of all filters.